learningequality/kolibri-design-system
# Fix outlines for KTextbox and KSwitch

## Description



#### Issue addressed


Addresses #131 

### Before/after screenshots


|`KTextbox` Before | `KTextbox` After| `KTextbox` (after background fixed) |
|--|--|--|
| (Basically, before there was no outline) | ![2021-05-28 15 33 54](https://user-images.githubusercontent.com/13563002/120047749-39442b80-bfca-11eb-839d-2bbe97ab50bc.gif)|![2021-06-08 23 40 21](https://user-images.githubusercontent.com/13563002/121306002-2aefec80-c8b3-11eb-95b9-407364b6e5f1.gif)|


|`KSwitch` Before | `KSwitch` After|
|--|--|
|![2021-05-28 13 33 34](https://user-images.githubusercontent.com/13563002/120046602-60e5c480-bfc7-11eb-8c11-6e4f17e4cdf6.gif)|![2021-05-28 15 07 25](https://user-images.githubusercontent.com/13563002/120046301-b79ece80-bfc6-11eb-8b72-e28270861a9a.gif)|


## Steps to test

`KTextbox`
1. On the sign-in page, tab until you get into the username textbox.
2. Check to see that there is an outline around the textbox.

`KSwitch`
1. Sign in as a coach
2. Click on "Reports" for your class
3. Under "Lessons," toggle the switch on/off for "Visible to learners"
4. Check to see if there is an outline that appears as you are toggling.
5. Please check for an RTL language as well by changing the language in the address bar in your browser from `/en/` to `/ar/` for example.

## (optional) Implementation notes

### At a high level, how did you implement this?
For both of these components, I added `:style` and a ternary for whether or not there was a focus on the element. I also made updates to `trackInputModality.js` in order to differentiate between keyboard tab and mouse click.



### Does this introduce any tech-debt items?


This in itself does not introduce any tech-debt items. However, it should be noted that there is a new CSS pseudo-class called `:focus-visible` that simplifies a lot of the logic that was introduced in order to track the use of keyboard vs. mouse, and it will likely be a refactor that we will want to look into in the design system. I will be opening an issue and linking this issue.

## Reviewer guidance




## Comments


I accidentally forgot and added both `KTextbox` and `KSwitch` together in this PR 😩
